### PR TITLE
fix some security issues

### PR DESCRIPTION
**Pull request by steelgeek091**

### CODE REVIEW

File: crates/rooch-rpc-server/src/server/rooch_server.rs [hunk 1]
```diff
@@ -393,24 +395,48 @@ impl RoochAPIServer for RoochServer {
...
+            let gaps = end
+                .checked_sub(start)
+                .ok_or(jsonrpsee::core::Error::Custom(
+                    "end value is overflow".to_string(),
+                ))?;
+            if gaps > MAX_RESULT_AMOUNT {
+                return Err(jsonrpsee::core::Error::Custom(
+                    "end value is overflow".to_string(),
+                ));
+            }
```
Comment (jolestar) [inline crates/rooch-rpc-server/src/server/rooch_server.rs:411]: We just need to check the limit argument is not too big.

Comment (jolestar) [reply crates/rooch-rpc-server/src/server/rooch_server.rs:411]: And `10000000` is too big, maybe `10000`?

File: crates/rooch-store/src/transaction_store/mod.rs [hunk 2]
```diff
@@ -35,6 +37,14 @@ pub trait TransactionStore {
         let start = cursor.unwrap_or(0);
         let end = start + limit;
 
+        let gaps = match end.checked_sub(start) {
+            None => return Err(anyhow::Error::msg("end value is overflow")),
+            Some(v) => v,
+        };
+        if gaps > MAX_RESULT_AMOUNT {
+            return Err(anyhow::Error::msg("end value is overflow"));
+        }
```
Comment (jolestar) [inline crates/rooch-store/src/transaction_store/mod.rs:46]: We do not need to check it in the internal storage; check the RPC argument is enough.

File: moveos/moveos-store/src/event_store/mod.rs [hunk 3]
```diff
@@ -154,15 +156,42 @@ impl EventDBStore {
...
             let start = cursor.unwrap_or(last_seq + 1);
             let end = if start >= limit { start - limit } else { 0 };
+
+            let gaps = match end.checked_sub(start) {
+                None => return Err(anyhow::Error::msg("end value is overflow")),
+                Some(v) => v,
+            };
+            if gaps > MAX_RESULT_AMOUNT {
+                return Err(anyhow::Error::msg("end value is overflow"));
+            }
```
Comment (jolestar) [inline moveos/moveos-store/src/event_store/mod.rs:166]: Same as above

File: crates/rooch-rpc-server/src/server/rooch_server.rs (hunk 1 above)
Comment (steelgeek091) [reply crates/rooch-rpc-server/src/server/rooch_server.rs:411]: Fixed

File: crates/rooch-store/src/transaction_store/mod.rs (hunk 2 above)
Comment (steelgeek091) [reply crates/rooch-store/src/transaction_store/mod.rs:46]: Fixed

File: moveos/moveos-store/src/event_store/mod.rs (hunk 3 above)
Comment (steelgeek091) [reply moveos/moveos-store/src/event_store/mod.rs:166]: This function is called by the `get_events_by_event_handle()` function in the `server/rooch_server.rs` file, but the code `(start..end).collect::<Vec<_>>()` is only executed internally within this function.

File: crates/rooch-rpc-server/src/server/rooch_server.rs (hunk 1 above)
Comment (baichuan3) [reply crates/rooch-rpc-server/src/server/rooch_server.rs]:
```        let limit_of = min(
            limit.map(Into::into).unwrap_or(DEFAULT_RESULT_LIMIT_USIZE),
            MAX_RESULT_LIMIT_USIZE,
        );
```
It is enough to add this check to `get_transactions_by_order`, and other overflow judgments about the limit are not needed.
Other methods are the same as above.

Comment (jolestar) [reply crates/rooch-rpc-server/src/server/rooch_server.rs]: Use the code of @baichuan3

**Comment by jolestar** [comment]
The event struct in examples need to be updated.

File: moveos/moveos-store/src/event_store/mod.rs (hunk 3 above)
Comment (jolestar) [reply moveos/moveos-store/src/event_store/mod.rs:166]: Check the limit argument is enough.

File: crates/rooch-rpc-server/src/server/rooch_server.rs (hunk 1 above)
Comment (steelgeek091) [reply crates/rooch-rpc-server/src/server/rooch_server.rs]: Fixed

**Comment by steelgeek091** [comment]
> The event structs in examples need to be updated.

Fixed

File: moveos/moveos-store/src/event_store/mod.rs (hunk 3 above)
Comment (steelgeek091) [reply moveos/moveos-store/src/event_store/mod.rs]: Fixed

**Review by jolestar — APPROVED** [review]

**Pull request merged**